src/pages/DatasetRegistration.js

```diff
@@ -101,6 +101,59 @@ class DatasetRegistration extends Component {
       prev.dacList = dacs;
       return prev;
     });
+    if (!fp.isEmpty(this.state.updateDataset)) {
+      this.prefillDatasetFields(this.state.updateDataset);
+    }
+  };
+
+  async init() {
+    const { datasetId } = this.props.match.params;
+    let updateDataset = {};
+    // update dataset case
+    if (!fp.isNil(datasetId)) {
+      DataSet.getDataSetsByDatasetId(datasetId).then(data => {
+        updateDataset = data;
+        // redirect to blank form if dataset id is invalid or inaccessible
+        if (fp.isEmpty(updateDataset) || fp.isNil(updateDataset.dataSetId)) {
+          this.props.history.push('/dataset_registration');
+          ReactTooltip.rebuild();
+        }
+        else {
+          this.setState(prev => {
+            prev.updateDataset = updateDataset;
+          });
+        }
+      });
+    }
+  };
+
+  // fill out the form fields with old dataset properties if they already exist
+  prefillDatasetFields(dataset) {
+    let name = dataset.properties.find(p => p.propertyName === "Dataset Name");
+    let collectionId = dataset.properties.find(p => p.propertyName === "Sample Collection ID");
+    let dataType = dataset.properties.find(p => p.propertyName === "Data Type");
+    let species = dataset.properties.find(p => p.propertyName === "Species");
+    let phenotype = dataset.properties.find(p => p.propertyName === "Phenotype/Indication");
+    let nrParticipants = dataset.properties.find(p => p.propertyName === "# of participants");
+    let description = dataset.properties.find(p => p.propertyName === "Description");
+    let datasetRepoUrl = dataset.properties.find(p => p.propertyName === "dbGAP");
+    let researcher = dataset.properties.find(p => p.propertyName === "Data Depositor");
+    let pi = dataset.properties.find(p => p.propertyName === "Principal Investigator(PI)");
+
+    this.setState(prev => {
+      prev.datasetData.datasetName = name ? name.propertyValue : '';
+      prev.datasetData.collectionId = collectionId ? collectionId.propertyValue : '';
+      prev.datasetData.dataType = dataType ? dataType.propertyValue : '';
+      prev.datasetData.species = species ? species.propertyValue : '';
+      prev.datasetData.phenotype = phenotype ? phenotype.propertyValue : '';
+      prev.datasetData.nrParticipants = nrParticipants ? nrParticipants.propertyValue : '';
+      prev.datasetData.description = description ? description.propertyValue : '';
+      prev.datasetData.datasetRepoUrl = datasetRepoUrl ? datasetRepoUrl.propertyValue : '';
+      prev.datasetData.researcher = researcher ? researcher.propertyValue : '';
+      prev.datasetData.pi = pi ? pi.propertyValue : '';
+
+      return prev;
+    })
```

Comment:
[codacy nit:](https://app.codacy.com/gh/DataBiosphere/duos-ui/pullRequest?prid=6449150)
```suggestion
    });
```